pkg/resource/deploy/step_generator.go

```diff
@@ -1451,6 +1487,13 @@ func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets) ([]Step, error)
 		}
 	}
 
+	// We also need to delete all the new resources that we created/updated/samed if this is a destroy operation.
+	for i := len(sg.toDelete) - 1; i >= 0; i-- {
```

Comment:
Maybe worth calling out the deliberate reverse order here and why it's important.